openmc/arithmetic.py

```diff
@@ -868,14 +886,85 @@ def get_pandas_dataframe(self, datasize, summary=None):
 
         import pandas as pd
 
-        # Construct a sring representing the filter aggregation
-        aggregate_bin = '{0}('.format(self.aggregate_op)
-        aggregate_bin += ', '.join(map(str, self.bins)) + ')'
+        # Create NumPy array of the bin tuples for repeating / tiling
+        filter_bins = np.empty(self.num_bins, dtype=tuple)
+        for i, bin in enumerate(self.bins):
+            filter_bins[i] = bin
 
-        # Construct NumPy array of bin repeated for each element in dataframe
-        aggregate_bin_array = np.array([aggregate_bin])
-        aggregate_bin_array = np.repeat(aggregate_bin_array, datasize)
+        # Repeat and tile bins as needed for DataFrame
+        filter_bins = np.repeat(filter_bins, self.stride)
+        tile_factor = data_size / len(filter_bins)
+        filter_bins = np.tile(filter_bins, tile_factor)
 
-        # Construct Pandas DataFrame for the AggregateFilter
-        df = pd.DataFrame({self.type: aggregate_bin_array})
+        # Create DataFrame with aggregated bins
+        df = pd.DataFrame({self.type: filter_bins})
         return df
+
+    def can_merge(self, other):
+        """Determine if AggregateFilter can be merged with another.
+
+        Parameters
+        ----------
+        other : AggregateFilter
+            Filter to compare with
+
+        Returns
+        -------
+        bool
+            Whether the filter can be merged
+
+        """
+
+        if not isinstance(other, AggregateFilter):
+            return False
+
+        # Filters must be of the same type
+        elif self.type != other.type:
+            return False
+
+        # None of the bins in this filter should match in the other filter
+        for bin in self.bins:
+            if bin in other.bins:
+                return False
+
+        # None of the bins in the other filter should match in this filter
+        for bin in other.bins:
+            if bin in self.bins:
+                return False
```

Comment:
Is this second check redundant? It seems like matching bins would be guaranteed to be found in the check above.